xml_converter/src/xml_converter.cpp

```diff
@@ -64,83 +65,85 @@ void write_xml_file(string xml_filepath, map<string, Category>* marker_categorie
-    waypoint::Waypoint all_categories;
-    waypoint::Waypoint output_message;
-    std::set<int> map_ids;
-    std::set<string> proto_categories;
-    std::set<string> category_includes;
-
-    if (mkdir(proto_filepath.c_str(), 0700) == -1) {
+    if (mkdir(proto_filepath.c_str(), 0700) == -1 && errno != EEXIST) {
         cout << "Error making ./protobins" << endl;
         throw std::error_code();
     }
-
+    // Converts map<string, Category> to Waypoint::Categories
+    waypoint::Waypoint all_categories;
+    std::set<string> proto_categories;
     for (const auto& category : *marker_categories) {
         waypoint::Category proto_category = category.second.as_protobuf();
         all_categories.add_category()->CopyFrom(proto_category);
         create_proto_categories(proto_category, &proto_categories);
     }
-
+    // Converts vector<Parseable*> to Waypoint::Icon and Waypoint::Trail
+    waypoint::Waypoint proto_pois;
+    std::set<int> map_ids;
     for (const auto& parsed_poi : *parsed_pois) {
         if (parsed_poi->classname() == "POI") {
             Icon* icon = dynamic_cast<Icon*>(parsed_poi);
             waypoint::Icon poi = icon->as_protobuf();
             proto_pois.add_icon()->CopyFrom(poi);
             map_ids.insert(icon->map_id);
         }
-        if (parsed_poi->classname() == "Trail") {
+        else if (parsed_poi->classname() == "Trail") {
             Trail* trail = dynamic_cast<Trail*>(parsed_poi);
             waypoint::Trail poi = trail->as_protobuf();
             proto_pois.add_trail()->CopyFrom(poi);
             map_ids.insert(trail->map_id);
         }
     }
 
+    waypoint::Waypoint output_message;
+    std::set<string> category_includes;
     for (int map_id : map_ids) {
         ofstream outfile;
         string output_filepath = proto_filepath + "/" + to_string(map_id) + ".data";
         outfile.open(output_filepath, ios::out | ios::binary);
         output_message.CopyFrom(all_categories);
 
         for (int i = 0; i < proto_pois.icon_size(); i++) {
-            vector<string> split_categories = split(proto_pois.icon(i).category().name(), ".");
             if (proto_pois.icon(i).map_id() == map_id) {
-                auto category = proto_categories.find(split_categories[split_categories.size() - 1]);
+                vector<string> split_categories = split(proto_pois.icon(i).category().name(), ".");
+                auto category = proto_categories.find(lowercase(split_categories[split_categories.size() - 1]));
                 if (category == proto_categories.end()) {
                     cout << "Unknown MarkerCategory " << proto_pois.icon(i).category().name() << endl;
                 }
                 else {
-                    for (string split_category : split_categories) {
-                        category_includes.insert(split_category);
+                    // This loop adds the name of the category and all of it's parents
+                    string name;
```

Comment:
Is this string name just manually reassembling the name that was manually dissasembled earlier?
IE:
is `string name` equivalent to `proto_pois.icon(i).category().name()`  after going through this for loop?

Reply:
Yes. By the end of the loop they are equivalent. The name is in the format `a.b.c.d` and I want to make the set include 
```
{
     a
     a.b
     a.b.c
     a.b.c.d
} 
```

Reply:
Yep that makes sense.